src/transaction/scarguments.cpp

```diff
@@ -0,0 +1,49 @@
+#include "transaction/scarguments.h"
+
+#include "hex.h"
+
+SCArguments::SCArguments() : m_args()
+{};
+
+void SCArguments::add(std::string const &arg)
+{
+    std::string const val = util::stringToHex(arg);
+
+    m_args.emplace_back(val);
+}
+
+void SCArguments::add(BigUInt const &arg)
+{
+    std::string const val = arg.getHexValue();
+
+    m_args.emplace_back(val);
+}
+
+void SCArguments::add(Address const &arg)
+{
+    bytes const pk = arg.getPublicKey();
+
+    std::string pkHex(pk.begin(), pk.end());
+    pkHex = util::stringToHex(pkHex);
```

Comment:
Perhaps in the future we can add `toHex` (`toPubKey`) to `Address` class as well.

Reply:
Maybe even better, we could:

- Change bytes typedef to be a class
- Add `toHex` method to it ?